### DISCUSSION TITLE
How to add gain ripple and noise ripple for amplifiers？

### DISCUSSION THREAD

**Discussion by OptiTransim** [Telecominfraproject/oopt-gnpy#481] [General]
For example, how to set the gain ripple to vary within 1db and the noise figure ripple to vary within a certain range. When simulating with GNPy, the 'high_detail_model_example' amplifier type is used, but there is no corresponding change in the GSNR results.

**Comment by EstherLerouzic** [comment]
Dear @OptiTransim,
You can use the vectors as defined in std_medium_gain_advanced_config.json and define an EDFA that uses it. The ripples should be defined as a vector per frequency. If you want to vary the range of the ripples, you need to create different vectors with increased or decreased amplitudes. Note that a single value will not do the job.

I join the config as example (these are just fake examples).
You should see impact on GSNR if you try these files.

[eqpt_config.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13530097/eqpt_config.json)
[std_medium_gain_advanced_config.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13530098/std_medium_gain_advanced_config.json)
[std_medium_gain_advanced_config_no_ripple.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13530100/std_medium_gain_advanced_config_no_ripple.json)
[std_medium_gain_advanced_config_ripple_2.json](https://github.com/Telecominfraproject/oopt-gnpy/files/13530101/std_medium_gain_advanced_config_ripple_2.json)

Apart from the ripples, note that there is a pending correction concerning gain tilt documented here #471 

I hope this helps

Regards
